### DISCUSSION TITLE
Better File Preview | 更好的文件预览

### DISCUSSION THREAD

**Discussion by arvinxx** [lobehub/lobehub#3684] [General]
Better File Preview | 更好的文件预览

LobeChat currently supports basic file preview functionality, but there's room for improvement in terms of user experience and supported file types. We're opening this discussion to gather feedback on enhancing our file preview capabilities.

LobeChat 目前支持基本的文件预览功能,但在用户体验和支持的文件类型方面还有提升空间。我们开放此贴收集关于改进文件预览功能的反馈。

---

Currently, the file preview features supported by LobeChat include:

目前 LobeChat 支持的文件预览功能包括:
- [x] **Image** : JPG/PNG/BMP/GIF
  - https://github.com/lobehub/lobe-chat/pull/3690
- [ ] Audio
- [ ] Video 

----

> [!IMPORTANT]
>
> If you have any suggestions for improving file preview or requests for new file type support, please comment below. 
>
> 如果您对改进文件预览或支持新文件类型有任何建议,请在下方评论。

### PRs Welcome 💞

- A demo PR of adding new file type Preview: https://github.com/lobehub/lobe-chat/pull/3690

**Comment by xiaoxuan220** [comment]
pls support json file preview

**Reply by 0x5t4l1n** [reply]
ok what did you need

**Comment by Alice39s** [comment]
pls support code files such as Typescript, thx!

**Reply by 0x5t4l1n** [reply]
ok how can i help you

**Comment by wyd-David** [comment]
please support xlsx and csv,thx!

**Reply by 0x5t4l1n** [reply]
ok

**Reply by arvinxx** [reply]
@wyd-David CSV is supported now

**Comment by moly7x** [comment]
please support text file like txt or md

**Reply by arvinxx** [reply]
but how do we know what's the text file?

**Reply by 0x5t4l1n** [reply]
.txt format file is texxt file

**Comment by hedeqiang** [comment]
.yaml

**Comment by dab512** [comment]
Please add mp3 and mp4 via media player

**Reply by arvinxx** [reply]
good idea

**Comment by peerless-hero** [comment]
.json and .json5

**Comment by hi-fox** [comment]
.json
.py
.pyw
.md
.html
.css
.js

**Comment by stevendong** [comment]
I noticed that the project code in src/features/FileViewer/index.tsx utilizes the library @cyntler/react-doc-viewer for file previews. According to the documentation of this library, it supports previewing the following file formats: bmp, csv, odt, doc, docx, gif, htm, html, jpg, jpeg, pdf, png, ppt, pptx, tiff, txt, xls, xlsx, mp4, and webp. However, the code only uses this library for image and text previews. I would like to ask what considerations led to the decision not to use the MP4 video format for previewing?

**Reply by arvinxx** [reply]
Because I prefer to add the preview support with user demand.Also some easy cases can be done by community.

PR welcome😆

**Comment by zzeitt** [comment]
.epub
